src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -367,29 +374,117 @@ private void RefreshBackupFileList(IEnumerable<string> filePaths)
+
                     var extension = Path.GetExtension(filePath).ToUpper();
                     // If not extension specified and code reach here, this means this is still a valid file
                     // only without file type. Otherwise, simply take extension substring skipping the 'dot'.
                     var subScript = extension.IndexOf(".") == 0 ? extension.Substring(1) : "";
                     var caption = Path.GetFileNameWithoutExtension(filePath);
 
+                    // Measure the performance for this region
+                    // deserializes the file only once
+                    var jsonObject = DeserializeJsonFile(filePath); 
+                    var description = jsonObject != null ? GetGraphDescription(jsonObject) : string.Empty;
+                    var thumbnail = jsonObject != null ? GetGraphThumbnail(jsonObject) : string.Empty;
+                    var author = jsonObject != null ? GetGraphAuthor(jsonObject) : Resources.DynamoXmlFileFormat;
+
+                    var date = GetDateModified(filePath);
+
                     files.Add(new StartPageListItem(caption)
                     {
                         ContextData = filePath,
                         ToolTip = filePath,
                         SubScript = subScript,
-                        ClickAction = StartPageListItem.Action.FilePath
-                    });
+                        Description = description,
+                        Thumbnail = thumbnail,
+                        Author = author,
+                        DateModified = date,
+                        ClickAction = StartPageListItem.Action.FilePath,
+
+                    }); 
                 }
                 catch (ArgumentException ex)
                 {
                     DynamoViewModel.Model.Logger.Log("File path is not valid: " + ex.StackTrace);
                 }
+                catch (Exception ex)
+                {
+                    DynamoViewModel.Model.Logger.Log("Error loading the file: " + ex.StackTrace);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Utility method to get the last time a file has been modified
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        private string GetDateModified(string filePath)
+        {
+            FileInfo fileInfo = new FileInfo(filePath);
+
+            if (fileInfo.Exists)
+            {
+                DateTime lastModified = fileInfo.LastWriteTime;
+                return lastModified.ToString();
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        private Dictionary<string, object> DeserializeJsonFile(string filePath)
+        {
+            try
+            {
+                var jsonString = File.ReadAllText(filePath);
+                return JsonConvert.DeserializeObject<Dictionary<string, object>>(jsonString);
```

Comment:
You can use `DynamoUtilities.PathHelper.isValidJson(filePath, out fileContents, out ex)`